nni/retiarii/oneshot/pytorch/utils.py

```diff
@@ -180,3 +180,117 @@ def replace_input_choice(root_module, init_fn, modules=None):
         A list from layer choice keys (names) and replaced modules.
     """
     return _replace_module_with_type(root_module, init_fn, (InputChoice, nn.InputChoice), modules)
+
+
+class ParallelTrainValDataLoader(DataLoader):
+    """
+    Dataloader that yields both train data and validation data in a batch, with an order of (train_batch, val_batch). The shorter
+    one will be upsampled (repeated) to the length of the longer one, and the tail of the last repeat will be dropped.
+
+    Some NAS algorithms, i.e. DARTS and Proxyless, require this type of dataloader.
```

Comment:
the name is "Parallel", does enas need this dataloader?

Reply:
no, ENAS uses `ConcatenateTrainValDataloader`